### ISSUE TITLE
Tetris Party Deluxe black screen freeze after playing any game and quitting to menu

### ISSUE THREAD

**Issue by clemwo** [DS-Homebrew/nds-bootstrap#1261]
**Your console:** 2DS & 2DS XL (this issue happens on 2 different 2DS XL and 2 different 2DS)
**Launch Method:** Launched via SD card
**Version used:** Bootstrap v0.46.1, TM++ v21.4.0
**Game tested/used/applicable:** Tetris Party Deluxe (USA) (En,Fr,Es), MD5Sum: 7da7954fce54b106be48684241fd4bcf 

**Expected Behavior:** Game doesn't freeze in blackscreen after quitting playing
**Actual Behavior:** Game freezes in blackscreen after playing any game mode of Tetris and then choosing "Quit" to go back to menu

### Steps to reproduce

Start singleplayer tetris game like marathon. Drop all the pieces quickly so that you lose. Then choose quit to go back to menu and game freezes on black screen.

bootstrap.ini:
```
[NDS-BOOTSTRAP]
DEBUG = 0
LOGGING = 0
CACHE_FAT_TABLE = 0
ROMREAD_LED = 0
DMA_ROMREAD_LED = -1
CONSOLE_MODEL = 2
NDS_PATH = sd:/ds roms/Tetris Party Deluxe (USA).nds
SAV_PATH = sd:/ds roms/saves/Tetris Party Deluxe (USA).sav
AP_FIX_PATH = sd:/_nds/nds-bootstrap/apFix.ips
RAM_DRIVE_PATH = sd:/null.img
GUI_LANGUAGE = en
LANGUAGE = -1
REGION = -2
DSI_MODE = 1
BOOST_CPU = 0
BOOST_VRAM = 0
CARD_READ_DMA = 1
ASYNC_CARD_READ = 0
EXTENDED_MEMORY = 0
DONOR_SDK_VER = 0
PATCH_MPU_REGION = 0
PATCH_MPU_SIZE = 0
FORCE_SLEEP_PATCH = 0
PRECISE_VOLUME_CONTROL = 0
SDNAND = 0
MACRO_MODE = 0
SOUND_FREQ = 0
HOTKEY = 284
```

### Logs

```
findModuleParamsOffset:
Module params offset restored: 02000BA4
sdk_version: 05037535
arm9 is compressed

Switched to NTR mode BIOS
relocate_ce9
firstCardLocation 023F4000 : 02780000

armReadCardLocation 023F40F0 : 02780000

thumbReadCardLocation 023F4110 : 02780000

armReadCardDmaLocation 023F4140 : 02780000

armReadSlot2Location 023F4164 : 02780000

armSetDmaCardLocation 023F4194 : 02780000

thumbReadCardDmaLocation 023F41C0 : 02780000

thumbSetDmaCardLocation 023F41E0 : 02780000

armReadNandLocation 023F4204 : 02780000

thumbReadNandLocation 023F4228 : 02780000

armWriteNandLocation 023F4240 : 02780000

thumbWriteNandLocation 023F4260 : 02780000

armIrqEnableLocation 023F4290 : 02780000

thumbIrqEnableLocation 023F42B0 : 02780000

pdashReadLocation 023F42DC : 02780000

thumbReadSlot2Location 023F4300 : 02780000

ipcSyncHandlerLocation 023F4368 : 02780000

resetLocation 023F4390 : 02780000

globalCardLocation 023F54E0 : 02780000

 ce9->patches 023F4034

patchCardNds

[SDK 5]

cardIrqEnable location : 020099B0

startOffset : 02000000

cardRead location : 020159C4
023F4000

swi12 location : 02000782

cardPullOut location : 02015F78

cardId location : 02015824


patchCardSetDma

NDMA CARD SET METHOD ACTIVE
cardSetDma location : 02015B80


NDMA CARD READ METHOD ACTIVE
cardEndReadDma location : 02015C10

findMpuInitCacheOffset:
Mpu init cache not found

patchMpuSize: 00000000

Mpu init: 0200B7FC

ERR_NONE

Relocation src: 023802E4

Arm7 (patch v5)
JumpTableFunc: 0238ECA8
card id:	0238EC90
card read:	0238ECA0
Eeprom read:	0238ECB4
Eeprom page write:	0238ECCC
Eeprom page prog:	0238ECE4
Eeprom verify:	0238ECFC
Eeprom page erase:	0238ED10
swiHalt location : 02380664

swi12 location : 02383600

swiGetPitchTable location : 02383614

ERR_NONE

hookNdsRetailArm7
hookLocation arm7: 0238FDBC

ERR_NONE
hookLocation arm9: 027E0000

ROM loaded into RAM

AP-fix found
AP-fix applied
chipID: 80000FC2
```

### Other notes
The game worked perfectly fine in nds-bootstrap v0.35.1 TM++ v16.0.0, only after upgrading the issues arose

**Comment by RocketRobz** [comment]
Have you tried turning off card read DMA?

**Comment by clemwo** [comment]
> Have you tried turning off card read DMA?

Just tried it, issue is still there unfortunately.

**Comment by RocketRobz** [comment]
Downgrade nds-bootstrap, and then test with each next version to see which version the issue started occurring in.

**Comment by clemwo** [comment]
> Downgrade nds-bootstrap, and then test with each next version to see which version the issue started occurring in.

0.43.2 works fine, issue starts appearing at 0.44

Thank you so much already for the support!

**Comment by RocketRobz** [comment]
Hmm, the only ideas I can think of as to why it crashes, is either the cloneboot fix or the user settings address patch being removed, though iirc, the same issue happened with the patch on my flashcard as well.

Try manually AP-patching the ROM.

**Comment by RocketRobz** [comment]
Possibly a duplicate of #1109

**Comment by RocketRobz** [comment]
Duplicate of #1109
